compiler/noirc_frontend/src/hir_def/types/arithmetic.rs

```diff
@@ -125,7 +125,7 @@ impl Type {
 
         // Note that this is exact, syntactic equality, not unification.
         // `rhs` is expected to already be in canonical form.
-        if l_op.inverse() != Some(op) || l_rhs.canonicalize() != *rhs {
+        if l_op.approx_inverse() != Some(op) || l_rhs.canonicalize() != *rhs {
```

Comment:
Should we emit a warning in the cases where `approx_inverse` is used?

Reply:
No, I don't think a warning is needed. We should limit uses of approx_inverse to cases where we know it is correct, e.g. when we're checking if something divides evenly